sympy/solvers/diophantine/diophantine.py

```diff
@@ -566,8 +586,10 @@ def solve(self, parameters=None, limit=None) -> DiophantineSolutionSet:
 
                     for z0 in range(0, abs(_c)):
                         # Check if the coefficients of y and x obtained are integers or not
-                        if (divisible(sqa*g*z0**2 + D*z0 + sqa*F, _c) and
-                            divisible(e*sqc*g*z0**2 + E*z0 + e*sqc*F, _c)):
+                        if (
+                            divisible(sqa*g*z0**2 + D*z0 + sqa*F, _c)
+                            and divisible(e*sqc*g*z0**2 + E*z0 + e*sqc*F, _c)
+                        ):
```

Comment:
I don't understand why both black and yapf seem to prefer the unindented reverse smiley `):`.

Reply:
I think the issue page on which you commented recently at black mentioned about how they tokenize and without looking ahead or back, that is just the simplest thing to do. It also visually marks the range of the `if`

Reply:
> I don't understand why both black and yapf seem to prefer the unindented reverse smiley `):`.

Haha reverse smiley. Actually, this is not the default yapf behaviour. It's a consequence of this option, which I added to mimic black:
https://github.com/sympy/sympy/blob/2d90dc1d91b1d16679bec2c0f94d49b836b5315b/setup.cfg#L95-L96

If you remove that option, you get:

```python
                        if (divisible(sqa*g*z0**2 + D*z0 + sqa*F, _c) and
                                divisible(e*sqc*g*z0**2 + E*z0 + e*sqc*F, _c)):
                            result.add((solve_x(z0), solve_y(z0)))
```

Even if the above is more 'concise', I think the split 'black' version is easier to understand at one glance.

Note: I had hoped `split_before_logical_operator = True` may improve the above a bit, but this seems to have no effect.

Reply:
> If you remove that option, you get:
> 
> ```python
>                         if (divisible(sqa*g*z0**2 + D*z0 + sqa*F, _c) and
>                                 divisible(e*sqc*g*z0**2 + E*z0 + e*sqc*F, _c)):
>                             result.add((solve_x(z0), solve_y(z0)))
> ```

That looks more like what I would have done when editing the code myself.